backend/infrahub/git/tasks.py

```diff
@@ -161,3 +162,82 @@ async def generate_artifact(model: RequestArtifactGenerate) -> None:
         log.exception("Failed to generate artifact", error=exc)
         artifact.status.value = "Error"
         await artifact.save()
+
+
+@flow(name="artifact-definition-generate")
+async def generate_request_artifact_definition(model: RequestArtifactDefinitionGenerate) -> None:
+    log.info(
+        "Received request to generate artifacts for an artifact_definition",
```

Comment:
would be good to add the function `add_branch_tag` in the flow
it was introduced in https://github.com/opsmill/infrahub/pull/4694
```
from infrahub.workflows.utils import add_branch_tag
await add_branch_tag(branch_name=model.branch)
```